### PR TITLE
Provide the ability to reset Fail2Ban count and ban flag

### LINKED ISSUE #113: Ability to reset specific throttles
I am finding it very difficult to do the following, which makes me think I'm using this software incorrectly or for the wrong use-case:

I am currently throttling login attempts to my app. When a user resets their password, I would like to reset the throttle for that specific user. It seems as if the Rack::Attack API is very unfriendly to such under-the-cover shenanigans. There is no throttle-specific reset function, nor is it particularly easy to recreate the key used to store the incrementer to allow for manual resetting.

What is the right way to go about this? Thanks!

### PR DESCRIPTION
**Pull request by stanhu**
This code is in production as a monkey-patch in GitLab v7.10, and I thought it would be helpful for others.

Closes #113

### CODE REVIEW

**Comment by stanhu** [comment]
@ktheory Thoughts?

File: lib/rack/attack/cache.rb [hunk 1]
```diff
@@ -30,6 +30,18 @@ def write(unprefixed_key, value, expires_in)
         store.write("#{prefix}:#{unprefixed_key}", value, :expires_in => expires_in)
       end
 
+      def reset_count(unprefixed_key, period)
+        epoch_time = Time.now.to_i
+        # Add 1 to expires_in to avoid timing error: http://git.io/i1PHXA
+        expires_in = period - (epoch_time % period) + 1
+        key = "#{(epoch_time / period).to_i}:#{unprefixed_key}"
```
Comment (ktheory) [inline lib/rack/attack/cache.rb]: @stanhu this nearly a dupe of lines 18-21. What do you think about extracting it to a private method?

**Comment by ktheory** [comment]
@stanhu: thanks for the nudge. :smile: 

This feature looks useful & I'd like to merge it in.

Could you please add some tests for the `Cache#reset_count` and `Cache#delete` methods? I want to make sure those work across the [various cache stores](https://github.com/kickstarter/rack-attack/blob/919976cb640c79163796b1e9290165e58f2c7748/spec/integration/rack_attack_cache_spec.rb#L19-L26).

**Comment by stanhu** [comment]
@ktheory: Thanks, done. There was already a test for `delete`, although this monkey patch seems necessary for the test only?

https://github.com/kickstarter/rack-attack/blob/919976cb640c79163796b1e9290165e58f2c7748/spec/integration/rack_attack_cache_spec.rb#L4

**Comment by ktheory** [comment]
@stanhu Awesome! Thanks so much. I aim to release version `4.3.0` with this change this afternoon.

**Comment by ktheory** [comment]
@stanhu Just pushed v4.3.0. Thanks for all your improvements recently. You're list release's MVP. :smile:

**Comment by stanhu** [comment]
@ktheory Thank you for being so responsive and creating/maintaining Rack Attack! I'm very happy I can remove the monkey patches now. :)

**Pull request merged**